Add execution locality option to backup docs (#16738)
cockroachdb/docs: v23.1/backup.md

CockroachDB's `BACKUP` [statement](sql-statements.html) allows you to create [full or incremental backups](take-full-and-incremental-backups.html) of your cluster's schema and data that are consistent as of a given timestamp.

> You can [backup a full cluster](#back-up-a-cluster), which includes:

Reviewer: ```suggestion
You can [back up a full cluster](#back-up-a-cluster), which includes:
```
The verb form :)

Author: Phew, nice catch!!!

Target                             | Description
-----------------------------------+-------------------------------------------------------------------------

> N/A                                | Backup the cluster. For an example of a full cluster backup, [see Backup a cluster](#back-up-a-cluster).

Reviewer: ```suggestion
N/A                                | Back up the cluster. For an example of a full cluster backup, [see Back Up a Cluster](#back-up-a-cluster).
```

> `DATABASE {database_name} [, ...]` | The name of the database(s) you want to backup (i.e., create backups of all tables and views in the database). For an example of backing up a database, see [Backup a database](#back-up-a-database).

Reviewer: ```suggestion
`DATABASE {database_name} [, ...]` | The names of the databases to back up. A database backup includes all tables and views in the database. Refer to [Back Up a Database](#back-up-a-database).
```

> `TABLE {table_name} [, ...]`       | The name of the table(s) or [view(s)](views.html) you want to backup. For an example of backing up a table or view, see [Backup a table or view](#back-up-a-table-or-view).

Reviewer: ```suggestion
`TABLE {table_name} [, ...]`       | The names of the tables and [views](views.html) to back up. Refer to [Back Up a Table or View](#back-up-a-table-or-view).
```
Worth linking tables as well as views?

> We recommend using the `BACKUP ... INTO {collectionURI}` syntax as per the following examples.

Reviewer: ```suggestion
Cockroach Labs recommends using the `BACKUP ... INTO {collectionURI}` syntax shown in following examples.
```

BACKUP bank.customers, bank.accounts INTO 'external://backup_s3' AS OF SYSTEM TIME '-10s';
~~~

### Back up all tables in a schema

>  To back up all tables in a [specified schema](create-schema.html), use a wildcard with the schema name:

Reviewer: ```suggestion
 To back up all tables in a [schema](create-schema.html), use a wildcard (`*`) with the schema name:
```

(1 row)
~~~

> ### Restrict nodes executing a backup job by locality

Author: Thoughts on the title here? "Restrict" may be negative? Perhaps, "control" or something similar.
